### PR TITLE
feat(WEBRTC-1615): Adding network_metrics event and monitor UI

### PR DESCRIPTION
**Pull request by DeividVeloso**
We add the new event to get `network_metrics` 

```ts
        roomRef.current.on(
          'network_metrics_report',
          (networkMetrics) => {
            console.log(
              'network_metrics_changed',
              networkMetrics
            );

            setNetworkMetrics(networkMetrics);
          }
        );
      }
```

 - Add `room.enableNetworkMetricsReport(participantIds, { includeStreams: true });` to start to get metrics
 - Add `room.disableNetworkMetricsReport()` to stop to get metrics
 
 
 
image

### CODE REVIEW

File: components/Feeds.tsx [hunk 1]
```diff
@@ -23,7 +25,19 @@ function Feeds({
   getParticipantStream: TelnyxRoom['getParticipantStream'];
   getStatsForParticipantStream: TelnyxRoom['getWebRTCStatsForStream'];
   dataTestId: string;
+  connectionQualityLevel: Map<string, NetworkMetrics>;
+  enableNetworkMetrics: any;
 }) {
+  useEffect(() => {
```
Comment (DeividVeloso) [inline components/Feeds.tsx:31]: This block of code should be removed before merge

File: components/Room.tsx [hunk 2]
```diff
@@ -89,7 +91,10 @@ function Room({
           )}
 
           {state.status === 'connected' && (
+            <div>
+              <button onClick={() => room.disableNetworkMetrics()}>STOP Metrics</button>
```
Comment (DeividVeloso) [inline components/Room.tsx:95]: This button should be removed before merge

File: components/Room.tsx [hunk 3]
```diff
@@ -89,6 +96,9 @@ function Room({
           )}
 
           {state.status === 'connected' && (
+            <div>
```
Comment (DeividVeloso) [inline components/Room.tsx:99]: Should remove before merge

File: hooks/room.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ import {
 
 import { DebugContext } from 'contexts/DebugContext';
 import { TelnyxMeetContext } from 'contexts/TelnyxMeetContext';
+import { NetworkMetrics } from '@telnyx/video/lib/metrics/interfaces';
```
Comment (arghav) [inline hooks/room.ts]: Let's export this from `@telnyx/video`

File: hooks/room.ts [hunk 5]
```diff
@@ -306,6 +310,12 @@ export const useRoom = ({
             });
           }
         );
+
+        roomRef.current.on('network_metrics_changed', (networkMetrics) => {
```
Comment (arghav) [inline hooks/room.ts:314]: Maybe we can rename this event to `network_metrics_report`?

File: components/WebRTCStats.tsx [hunk 6]
```diff
@@ -5,7 +5,7 @@ const ContainerOverlay = styled.div`
   position: absolute;
   width: 100%;
   height: 100%;
-  z-index: 1;
+  z-index: 5;
```
Comment (arghav) [inline components/WebRTCStats.tsx:8]: Let's set this back to `1`

File: components/Room.tsx [hunk 7]
```diff
@@ -38,6 +38,8 @@ function Room({
     },
   });
 
+
+
```
Comment (arghav) [inline components/Room.tsx:42]: Prettier needs to be configured to format this automatically

File: components/Room.tsx [hunk 8]
```diff
@@ -89,6 +96,9 @@ function Room({
           )}
 
           {state.status === 'connected' && (
+            <div>
+              <button onClick={() => room.enableNetworkMetrics(participantIds)}>Start Metrics</button>
+              <button onClick={() => room.disableNetworkMetrics()}>STOP Metrics</button>
             <Feeds
```
Comment (arghav) [inline components/Room.tsx:102]: Let's move this to `useRoom` hook

File: components/Feed.tsx [hunk 9]
```diff
@@ -24,13 +24,15 @@ function Feed({
...
   dataId,
+  networkMetrics,
 }: {
   participant: Participant;
   stream?: Stream;
   isSpeaking: boolean;
   getStatsForParticipantStream: TelnyxRoom['getWebRTCStatsForStream'];
   mirrorVideo: boolean;
   dataId?: string;
+  networkMetrics: TelnyxRoom['networkMetrics']
```
Comment (arghav) [inline components/Feed.tsx:35]: Let's remove this and use the context

**Review by arghav — CHANGES_REQUESTED** [review]

**Review by arghav — APPROVED** [review]
LGTM! 🎉

**Review by muriware — APPROVED** [review]
LGTM!

**Pull request merged**